programs/cleos/main.cpp

```diff
@@ -2957,6 +2961,201 @@ int main( int argc, char** argv ) {
       std::cout << fc::json::to_pretty_string(unpacked_action_data_json) << std::endl;
    });
 
+   // pack hex
+   string json;
+   string type;
+   string abi_file;
+   auto pack_hex = convert->add_subcommand("pack_hex", localized("From JSON to packed HEX form"));
+   pack_hex->add_option("json", json, localized("The JSON of built-in types including: signed_block, transaction/action_trace, transaction, action, abi_def"))->required();
+   pack_hex->add_option("--type", type, (localized("Type of the JSON data")))->required();
+   pack_hex->add_option("--abi-file", abi_file, localized("The abi file that contains --type for packing"));
+   pack_hex->callback([&] {
+      fc::variant unpacked_json = json_from_file_or_string(json);
+
+      std::cerr << "Packing: " << fc::json::to_pretty_string(unpacked_json) << "\n" << std::endl;
+      bool success = false;
+      std::optional<abi_def> abi;
+      if (!abi_file.empty()) {
+         abi = json::from_file(abi_file).as<abi_def>();
+      }
+      try {
+         abi_serializer s = abi ? abi_serializer(*abi, abi_serializer::create_yield_function( abi_serializer_max_time )) : abi_serializer{};
+         auto v = s.variant_to_binary(type, unpacked_json, fc::microseconds::maximum());
+         std::cout << fc::json::to_pretty_string(v) << std::endl;
+         success = true;
+      } catch (...) {}
+      if (!success) {
+         try {
+            if (type == "block_state_legacy") {
+               block_state_legacy t;
+               from_variant(unpacked_json, t);
+               auto v = fc::raw::pack<block_state_legacy>(t);
```

Comment:
I don't think you need to specify the template parameter (like  `<block_state_legacy>`) in these `pack()` calls.